File: backend/app/models/attendance.py
from sqlalchemy import Column, Integer, String, Date, Text, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.database import Base


class Attendance(Base):
    """考勤记录"""
    __tablename__ = "attendances"

    id = Column(Integer, primary_key=True, index=True)
    student_id = Column(Integer, ForeignKey("students.id", ondelete="CASCADE"), nullable=False, index=True)
    course_id = Column(Integer, ForeignKey("courses.id", ondelete="SET NULL"))
    attendance_date = Column(Date, nullable=False, index=True)
    status = Column(String(20), nullable=False)
    notes = Column(Text)

    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True))
    deleted_by = Column(Integer)

File: frontend/src/types/attendance.ts
export interface Attendance {
  id: number;
  student_id: number;
  student_name?: string;
  course_id?: number;
  course_name?: string;
  attendance_date: string;
  status: string;
  notes?: string;
  created_at: string;
  updated_at?: string;
}

export interface AttendanceCreate {
  student_id: number;
  course_id?: number;
  attendance_date: string;
  status: string;
  notes?: string;
}
